lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp

```diff
@@ -213,6 +213,8 @@ int amf_app_handle_deregistration_req(amf_ue_ngap_id_t ue_id) {
   // Remove stored UE context from AMF core.
   amf_remove_ue_context(ue_context);
 
+  amf_delete_ue_context(ue_context);
```

Comment:
Looks like `amf_remove_ue_context` is called within `amf_delete_ue_context`. You can remove line 214.

Reply:
+1 on this, I would encourage to follow the state manager's functions naming:
- For functions related with removing local state, use `free_ue_context` 
- For functions that delete stored state on redis, add `to_db` and `from_db` postfixes

Reply:
Addressed comments and renamed function name to amf_free_ue_context